### PR TITLE
Implement all arg types for `to_datetime`

### LINKED ISSUE #656: Support more arg types in to_datetime
cc @hendrikmakait 

test_to_datetime in dask/dask

### PR DESCRIPTION
**Pull request by hendrikmakait**
Closes #656

### CODE REVIEW

File: dask_expr/_collection.py [hunk 1]
```diff
@@ -3546,10 +3548,37 @@ def to_numeric(arg, errors="raise", downcast=None, meta=None):
...
+        elif not (is_dataframe_like(arg) or is_series_like(arg)):
+            raise NotImplementedError(
+                "dask.dataframe.to_datetime does not support "
+                "non-index-able arguments (like scalars)"
+            )
+        else:
+            meta = meta_series_constructor(arg)([pd.Timestamp("2000", **tz_kwarg)])
+            meta.index = meta.index.astype(arg.index.dtype)
+            meta.index.name = arg.index.name
+    if PANDAS_GE_200 and "infer_datetime_format" in kwargs:
```
Comment (phofl) [inline dask_expr/_collection.py:3569]: Lets deprecate in dask/dask, I don't want this stuff in here if we can avoid it

File: dask_expr/_collection.py [hunk 2]
```diff
@@ -3546,10 +3548,37 @@ def to_numeric(arg, errors="raise", downcast=None, meta=None):
...
+        warnings.warn(
+            "The argument 'infer_datetime_format' is deprecated and will be removed in a future version. "
+            "A strict version of it is now the default, see "
+            "https://pandas.pydata.org/pdeps/0004-consistent-to-datetime-parsing.html. "
+            "You can safely remove this argument.",
+            UserWarning,
+        )
+        kwargs.pop("infer_datetime_format")
+
+    # if not isinstance(arg, FrameBase):
```
Comment (phofl) [inline dask_expr/_collection.py:3579]: Is this intentional?

File: dask_expr/_collection.py (hunk 1 above)
Comment (phofl) [reply dask_expr/_collection.py:3569]: I'd also be ok with just ignoring it

File: dask_expr/_collection.py (hunk 2 above)
Comment (hendrikmakait) [reply dask_expr/_collection.py]: Whoops, removed.

File: dask_expr/_collection.py (hunk 1 above)
Comment (hendrikmakait) [reply dask_expr/_collection.py:3569]: This is already deprecated in `dask/dask`. I'll remove the handling.

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
thx

**Pull request merged**